RestCommClient/Classes/RCConnection.m

```diff
@@ -304,10 +301,36 @@ - (void)sipManagerDidReceiveOutgoingDeclined:(SipManager*)sipManager;
         [self.callingPlayer stop];
         self.callingPlayer.currentTime = 0.0;
     }
-    self.state = RCConnectionStateDisconnected;
     
-    [self.busyPlayer play];
+    // We want to tell to delegate that connection is declined immediately
+    // so, we need to play busy tone i new thread because we dont know what client will do
+    // for example, if set instance on connection to nil, the sound will not be played
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
```

Comment:
Nope, there is no concept like it in objc. There are some examples where category is created to simulate it; My opinion is that this will introduce confusion in the sdk.  The only thing I can think about is maybe to have singleton class which will playing tones independently from RCDevice? Of course if we don't want to handle this via delegate RCDevice / RCConnection.